generate_report_ui.gs

```diff
@@ -0,0 +1,263 @@
+    var reportUrl = CardService.newTextParagraph()
+    .setText("<b>Report Url</b>\n <a href='"+reportUrlVal+"'>"+reportUrlVal+"</a>");
+    section2.addWidget(reportUrl);
+  }
+    
+  // Create a button for generating report
+  var getReportAction = CardService.newAction()
+      .setFunctionName('generateReport'); 
+  var getReport = CardService.newTextButton()
+      .setText('Generate Report')
+      .setOnClickAction(getReportAction)
+      .setTextButtonStyle(CardService.TextButtonStyle.FILLED);
+  section2.addWidget(getReport);
+  card.addSection(section2);
+  
+   return card.build();
+}
+
+/**
+* Callback function on clicking the GET REPORT button and create a card with most executed script and most active users
+* @return {CardService.Card} The card to show to the user
+*/
+function generateReport(e) {
+  var title, projTitle;
+  var toTime = new Date();
+  var fromTime;
+  var millisInBetween;
+  var input  = e.formInput;
+  var projectFilter = input.projectFilter;
+  var reportType = input.reportType;
+  var limitResponse = input.limitResponse;
+  if(!limitResponse) limitResponse = 1;
+    
+  if(input.timeFilter == "LAST_HOUR") { 
+    title = "Last Hour";
+    millisInBetween = 60*60*1000;
+  } else if(input.timeFilter == "LAST_6_HOUR") {
+    title = "Last 6 Hours";
+    millisInBetween = 6*60*60*1000;
+  } else if(input.timeFilter == "LAST_24_HOUR") {
+    title = "Last 24 Hours";
+    millisInBetween = 24*60*60*1000;
+  } else if(input.timeFilter == "LAST_7_DAYS") {
+    title = "Last 7 Days";
+    millisInBetween = 7*24*60*60*1000;
+  } else {
+    title = "Last 30 Days";
+    millisInBetween = 30*24*60*60*1000;
+  }
+  
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    projTitle = "Specific Project";
+  } else if(projectFilter == "CUSTOM_PROJECT") {
+    projTitle = "Custom Projects";
+  } else if(projectFilter == "SYSTEM_PROJECT") {
+    projTitle = "System Projects";
+  } else {
+    projTitle = "All Projects";
+  }
+  
+  // variables to get the start time and end time
+  var fromTime = new Date(toTime.getTime() - millisInBetween);
+  var toTime = Utilities.formatDate(toTime, 'Etc/GMT', 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'');
+  var fromTime = Utilities.formatDate(fromTime, 'Etc/GMT', 'yyyy-MM-dd\'T\'HH:mm:ss.SSS\'Z\'');
+  
+  // if the selected option is specific project
+  if(projectFilter == "SPECIFIC_PROJECT") {
+    if(reportType == "mostExecutedScript") {
+      var processes = getNumberOfExecutionOfScript(input.projectId, fromTime);
+      var mostExecutedScript = convertObjectToSortedArrayForSpecificProject(processes, input.projectId);
+      var reportUrl = generateReportForMostExecutedScript(mostExecutedScript, "Report for Most Executed Script of "+ projTitle + " for " + title, limitResponse);
+    }
+    else if(reportType == "mostActiveUsers") {
+      var users = getUsersWithProcessId(input.projectId, fromTime);
+      var mostActiveUsers = convertObjectToSortedArrayForSPECIFIC_PROJECT(users, input.projectId);
```

Comment:
Fix the naming convention. Naming should be in camel case only

Reply:
Updated